### ISSUE TITLE
Anchors functioning as buttons missing button roles

### ISSUE THREAD

**Issue by aviannamiller** [omeka/omeka-s#2408]
Across both admin and the front-end, anchors are often used as buttons but lack button roles (and are thus communicating to screen readers as links). Either button roles should be added or anchor elements should be replaced with buttons.

**Comment by kimisgold** [comment]
I'm going to start a list of impacted areas so that we can track it. If I've missed anything, let me know and I'll update this.

- [x] Header
    - [ ] #user `url`
        - [ ] User ID (.user-show .o-icon-user .button) `url`
        - [ ] Edit User (.user-settings .button) `url`
        - [ ] Logout (.logout) `url`
    - [ ] nav#menu `url`
        - [ ] View public site (.o-icon-external .public .button) `url`
- [ ] Main content
    - [ ] #page-actions
        - [x] Make private / Make public (.button .o-icon-public / .o-icon-private)
            - [x] Item
            - [x] Item set
            - [x] Media
            - [x] Add site
            - [x] Add site page
            - [x] Edit site page
        - [x] Delete (.delete .button)
        - [x] Cancel (.cancel .button) `url`
        - [ ] Edit User (.button) `url`
        - [x] Add new Asset (.button .sidebar-content)
        - [ ] Import (.button) `url`
        - [ ] Add new resource template (.button) `url`
        - [ ] Import new vocabulary (.button) `url`
        - [ ] Reset (.button)
        - [x] Open / Close item set (.button .o-icon-lock / .o-icon-unlock)
        - [ ] Add new site (.button) `url`
        - [ ] Copy (.button .sidebar-content) `module content`
        - [ ] Edit `url`
        - [ ] View `url`
    - [ ] Form
        - [ ] .actions
            - [x] Edit (.o-icon-edit)
            - [x] Delete / Remove (.o-icon-delete)
            - [x] Undo (.o-icon-undo)
            - [ ] Copy (.far .fa-copy .sidebar-content) `module content`
            - [x] Details (.o-icon-more)
            - [x] Set language (.value-language .o-icon-language)
            - [x] More actions (.o-icon-more)
            - [x] Make private / Make public (.o-icon-public / .o-icon-private .value-visibility
                - [x] Resource form > manage media
                - [x] Resource form > value
                - [x] Resource form > value annotation
            - [x] Annotate value (.value-annotation-annotate .o-icon-annotation)
            - [x] Configure (.o-icon-settings)
            - [x] Expand / Collapse (.o-icon-expand .toggle-block-visibility .collapse / .expand)
        - [x] .add-values
            - [x] Text (.add-value .button .o-icon-literal)
            - [x] Omeka resource (.add-value .button .o-icon-resource)
            - [x] URI (.add-value .button .o-icon-uri)
        - [x] .inputs
            - [x] Add value (.add-value .button)
        - [x] .resource-inputs
            - [x] Items (.o-icon-items .button .resource-select)
            - [x] Item sets (.o-icon-item-sets .button .resource-select)
            - [x] Media (.o-icon-media .button .resource-select)
        - [ ] Honorable mentions:
            - [ ] .pagination-buttons (in spans, not anchors) `url`
                - [ ] .previous .o-icon-prev .button `url`
                - [ ] .next .o-icon-next .button `url`
    - [ ] Sidebars
        - [x] Close (.sidebar-close .o-icon-close)
        - [ ] View this item / item set (.o-icon-external) `url`
        - [x] Select resource (.button)
        - [x] .actions
            - [x] Make private / Make public (.o-icon-public / .o-icon-private .value-annotation-visibility
            - [x] Remove value (.o-icon-delete .value-annotation-remove)
            - [x] Restore value (.o-icon-undo .value-annotation-restore)
            - [x] Set language (.value-language .o-icon-language)
        - [x] Honorable mentions:
            - [x] .pagination-buttons (in spans, not anchors)
                - [x] .previous .o-icon-prev .button
                - [x] .next .o-icon-next .button
    - [x] Admin dashboard
        - [ ] Add site (.add .button) `url`
        - [ ] View public site (.o-icon-external .public .button) `url`
    - [x] Modules
        - [x] .module-actions
            - [ ] Configure (.o-icon-configure .button) `url`
            - [x] Uninstall (.o-icon-uninstall .button)
    - [x] Theme
        - [ ] Edit theme settings (.theme-settings .button) `url`
        - [ ] Configure resource pages (.theme-resource-pages .button) `url`
    - [ ] Navigation
        - [ ] All buttons within #nav-tree are presented as <i>, so none are properly coded

**Comment by kimisgold** [comment]
I've started work on this on the `anchors-to-buttons` branch, focusing on actions on the page editor as well as the public/private visibility toggles in `#page-actions` and within actions in the main content. I would appreciate early looks there to see if the markup looks solid and catch any implementation corrections early.

Quick question: do we actually want to convert the anchors that lead the user to another view? I'm thinking of elements like the anchors that lead to the user account, edit, or logout in the header, or delete and cancel actions in `#page-actions`.

**Comment by aviannamiller** [comment]
Thanks, I'll take a look! And that's a good point–let's only make the conversion to buttons if an action is triggered / action is happening on the same page. If it's directing elsewhere, it should stay an anchor. I do feel like logout and delete and cancel make sense as buttons since they're performing actions but open to differing thoughts here!

**Comment by aviannamiller** [comment]
External links should also remain anchors, then. That goes for view site / items / item set links, etc.

**Comment by aviannamiller** [comment]
On the page editor, some page blocks with expand / collapse sections involve the section heading in with the toggle, and it doesn't look those toggles have yet been converted to buttons. However, they have picked up the styles as they have the `.expand`/`.collapse` class, so they're now displaying inline:

`develop`:
<img width="703" height="510" alt="Page editor with 'Map by query' block. Toggled subheadings within block display in list form, styled in grey text" src="https://github.com/user-attachments/assets/4a9b832e-46f0-4e0a-b2bf-32d480aa1c16" />

`anchors-to-buttons`:
<img width="706" height="440" alt="Page editor with 'Map by query' block. Toggled subheadings within block display inline, styled in red" src="https://github.com/user-attachments/assets/561b7614-5a35-466e-ac5e-e14f037af918" />

This impacts Map by query, Asset, Image annotate asset, Image annotate media, Item Carousel, and Item with metadata blocks.

Toggles on Block layout config sidebars have also picked up some odd positioning:
<img width="314" height="550" alt="Block layout configuration sidebar. Toggles on subheading are pressed right against text labels." src="https://github.com/user-attachments/assets/1f4d439f-1712-44e8-a267-ff5882d66b69" />

Also on the topic of expand / collapse toggles, it would be great if the button names could incorporate the associated label element (whether that be a label within field-meta or the block heading). This is something we should think about generally for repeated buttons on a single page. See this excerpt from the VO Buttons rotor menu:

<img width="561" height="635" alt="VoiceOver Buttons rotor menu. Terms repeat in list, such as 'Expand collapsed button', 'Remove block button', and 'Configure layout button'." src="https://github.com/user-attachments/assets/0192d571-64d8-44ca-b134-ee8b1632e47d" />

Tying in those additional labels could really help users drill down to what they're looking for when there's a lot of information and repetitive controls on a page, such as on an edit form.

**Comment by kimisgold** [comment]
I think I've covered all the instances that apply and it's ready for a look. The only issues you've found I haven't responded to are the specificity of the expand collapse buttons and some of the alignments on the expand/collapse in Map by query, as that's module content. I've marked in the initial list the anchors that take the user to URLs.

**Comment by aviannamiller** [comment]
There still seem to be some anchors that haven't yet converted. I'll continue working through the list and confirm. In the meantime, I'm noticing the following:

- In resource browse views, copy functionality now doesn’t seem to be working. The copy confirm sidebars are not opening.
- Missing button text in a few areas, across all tested browsers:
  - Delete button page action on site edit page:
    <img width="659" height="387" alt="Site edit page. Delete button, which is missing inner contents, is highlighted by browser dev tools" src="https://github.com/user-attachments/assets/8cf70ba7-a5bd-4376-bc09-58c126b9fd5d" />
  - Add value button within resource forms:
    <img width="452" height="125" alt="Detail of resource edit form. Add Value button, which is missing inner contents, is highlighted by browser dev tools" src="https://github.com/user-attachments/assets/ba2c0b2e-218d-471b-af57-abe92edf6b72" />
There may be other spots with a similar issue, so I'll be sure to continue to check for that.

- A few times on this branch, I encountered what I presume to be an encoding issue with FA icons? Not sure why that might be and can’t consistently reproduce, but it’s happening sometimes after clicking through to a new page (item edit and resources, for instance):
  <img width="1253" height="578" alt="Admin-side resources page. Throughout the page, FA icons are replaced by strings of characters" src="https://github.com/user-attachments/assets/4e091bc8-7325-4d2f-b538-3ea441995c96" />
- Expand/collapse toggles are now styled in red [rgb(169, 25, 25)] everywhere. Will they remain this way or return to that dark grey tone? The red potentially feels a bit distracting in certain implementations (such as longer forms).

**Comment by aviannamiller** [comment]
I believe I have the list up to date so far, so I'm still noticing some anchors that should likely be replaced. Let me know if it would at all be helpful for me to cite specific instances within the list.

**Comment by kimisgold** [comment]
I think I've gotten the problem areas you've noticed. Please feel free to note specific instances I've overlooked, I think we're approaching the whack-a-mole stage of this issue.

**Comment by aviannamiller** [comment]
Fixes look great, thanks. Remaining issues where _I think_ elements should be converted to buttons are as follows:
- #page-actions
  - Reset (.button)
- .actions
  - Copy (.far .fa-copy .sidebar-content)
- .resource-inputs
  - Items (.o-icon-items .button .resource-select)
  - Item sets (.o-icon-item-sets .button .resource-select)
  - Media (.o-icon-media .button .resource-select)
- .pagination-buttons (in sidebars and forms)
  - .previous .o-icon-prev .button
  - .next .o-icon-next .button
- All icons on site Navigation #nav-tree

**Comment by kimisgold** [comment]
* Resource inputs and sidebar pagination have been converted.
* The "Reset" button leads to a URL so should remain an anchor.
* "Copy" is module content, so we'll get to that separately.
* I've done some work on the site navigation interface, but it's based on a third-party plugin. The plugin makes changing various parts of its markup somewhat difficult, and has several choices I disagree with. For instance, each node is an anchor element, and our current setup means that anchor element contains several buttons, including one that sends the user to a url in a new window. @zerocrates, would you have any thoughts on customizing the jstree default markup?

**Comment by aviannamiller** [comment]
Thanks for the changes, @kimisgold! I think we're almost there.

For the reset button, my thinking was that although it leads to a url, it triggers a page reload and stays on the same page, so there's no navigation elsewhere. The function here seems to hit a number of criteria for the use of a button, including "submits a form or user input". Would a button with `type="reset"` not then suffice here?

**Comment by kimisgold** [comment]
In offline discussions of the reset buttons and navigation, we've determined that they need their own issues due to the complexity of their use cases and implementations. Are there any other anchors that still need to be addressed?

**Comment by aviannamiller** [comment]
Within the scope of the list, I think we're about covered in this issue. Let's follow the rest in those issues and get to module content in another future pass.
